src/test/ensemble_flow_custom_node_tests.cpp

```diff
@@ -1140,6 +1140,177 @@ TEST_F(EnsembleFlowCustomNodeFactoryCreateThenExecuteTest, ParallelPipelineFacto
     }
 }
 
+struct AddSubInternalManager {
+    uint64_t* inputDims;
+    uint64_t* outputDims;
+    struct CustomNodeTensorInfo* inputInfo;
+    struct CustomNodeTensorInfo* outputInfo;
+    struct CustomNodeTensor* outputTensor;
+    uint8_t* outputTensorData;
+    uint64_t* outputTensorDims;
+
+    AddSubInternalManager() {
+        inputDims = (uint64_t*)malloc(2 * sizeof(uint64_t));
+        outputDims = (uint64_t*)malloc(2 * sizeof(uint64_t));
+        inputInfo = (struct CustomNodeTensorInfo*)malloc(1 * sizeof(struct CustomNodeTensorInfo));
+        outputInfo = (struct CustomNodeTensorInfo*)malloc(1 * sizeof(struct CustomNodeTensorInfo));
+        outputTensor = (struct CustomNodeTensor*)malloc(1 * sizeof(struct CustomNodeTensor));
+        outputTensorData = (uint8_t*)malloc(10 * 4 * sizeof(uint8_t));
+        outputTensorDims = (uint64_t*)malloc(2 * sizeof(uint64_t));
+    }
+
+    bool isPtrOwnedByManager(void* ptr) {
+        return (ptr == inputDims || ptr == outputDims || ptr == inputInfo || ptr == outputInfo || ptr == outputTensor || ptr == outputTensorData || ptr == outputTensorDims);
+    }
+};
+
+struct LibraryAddSubWithInternalManager {
+    static int initialize(void** customNodeLibraryInternalManager, const struct CustomNodeParam* params, int paramsCount) {
+        std::unique_ptr<AddSubInternalManager> internalManager = std::make_unique<AddSubInternalManager>();
+        *customNodeLibraryInternalManager = internalManager.release();
+        return 0;
+    }
+    static int deinitialize(void* customNodeLibraryInternalManager) {
+        if (customNodeLibraryInternalManager != nullptr) {
+            AddSubInternalManager* internalManager = static_cast<AddSubInternalManager*>(customNodeLibraryInternalManager);
```

Comment:
Just random thought, if user closes ovms with CTRL+C there is no cleanup of internal managers.

Reply:
I am not aware what is the expected behaviour in that situation, however I suppose that in order to cleanup internal managers pipelineDefinition and potentially CustomNode objects need to be destroyed. With introduction of cleanup mechanism additionally watcher thread will be responsible.